netbox/utilities/forms/mixins.py

```diff
@@ -75,3 +160,89 @@ class DistanceValidationMixin(forms.Form):
             MaxValueValidator(Decimal(100000)),
         ]
     )
+
+
+class FilterModifierMixin:
+    """
+    Mixin that enhances filter form fields with lookup modifier dropdowns.
+
+    Automatically detects fields that could benefit from multiple lookup options
+    and wraps their widgets with FilterModifierWidget.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._enhance_fields_with_modifiers()
+
+    def _enhance_fields_with_modifiers(self):
+        """Wrap compatible field widgets with FilterModifierWidget."""
+
+        model = getattr(self, 'model', None)
+        if model is None and hasattr(self, '_meta'):
+            model = getattr(self._meta, 'model', None)
+
+        filterset_class = None
+        if model:
+            key = f'{model._meta.app_label}.{model._meta.model_name}'
+            filterset_class = registry['filtersets'].get(key)
+
+        filterset = filterset_class() if filterset_class else None
+
+        for field_name, field in self.fields.items():
+            lookups = self._get_lookup_choices(field)
+
+            if filterset:
+                lookups = self._verify_lookups_with_filterset(field_name, lookups, filterset)
+
+            if len(lookups) > 1:
+                field.widget = FilterModifierWidget(
+                    widget=field.widget,
+                    lookups=lookups
+                )
+
+    def _get_lookup_choices(self, field):
+        """Determine the available lookup choices for a given field.
+
+        Returns an empty list for fields that should not be enhanced.
+        """
+        if isinstance(field, QueryField):
+            return []
+
+        if isinstance(field, (forms.BooleanField, forms.NullBooleanField)):
+            return []
+
+        if self._is_api_widget_field(field):
+            return []
+
+        for field_class in field.__class__.__mro__:
+            if field_class in FORM_FIELD_LOOKUPS:
+                return FORM_FIELD_LOOKUPS[field_class]
```

Comment:
One thing I've noticed is that the empty/not empty lookups are added even for mandatory model fields (e.g. `Site.status`). While not necessarily problematic, this can be confusing to the user. However, I don't see any direct path to glean that context via the model's filterset. Not sure it's worth tackling at this time.

Reply:
I took a stab at this, by trying to check the the field on form and then on the model. I didn't love it.